src/Explorer/Controls/Settings/SettingsComponent.tsx

```diff
@@ -0,0 +1,1403 @@
+  canThroughputExceedMaximumValue,
+  GeospatialConfigType,
+  TtlType,
+  ChangeFeedPolicyState,
+  SettingsV2TabTypes,
+  getTabTitle
+} from "./SettingsUtils";
+import { ConflictResolutionComponent } from "./SettingsSubComponents/ConflictResolutionComponent";
+import { SubSettingsComponent } from "./SettingsSubComponents/SubSettingsComponent";
+import { Pivot, PivotItem, IPivotProps, IPivotItemProps } from "office-ui-fabric-react";
+import "./SettingsComponent.less";
+import { IndexingPolicyComponent } from "./SettingsSubComponents/IndexingPolicyComponent";
+
+type StatefulValuesType = boolean | string | number | DataModels.IndexingPolicy;
+
+interface SettingsV2TabInfo {
+  tab: SettingsV2TabTypes;
+  content: JSX.Element;
+}
+
+interface ButtonV2 {
+  isVisible: () => boolean;
+  isEnabled: () => boolean;
+  isSelected?: () => boolean;
+}
+
+enum StatefulValueNames {
+  Throughput = "throughput",
+  TimeToLive = "timeToLive",
+  TimeToLiveSeconds = "timeToLiveSeconds",
+  GeospatialConfigType = "geospatialConfigType",
+  IndexingPolicyContent = "indexingPolicyContent",
+  ConflictResolutionPolicyMode = "conflictResolutionPolicyMode",
+  ConflictResolutionPolicyPath = "conflictResolutionPolicyPath",
+  ConflictResolutionPolicyProcedure = "conflictResolutionPolicyProcedure",
+  AnalyticalStorageTtlSelection = "analyticalStorageTtlSelection",
+  AnalyticalStorageTtlSeconds = "analyticalStorageTtlSeconds",
+  ChangeFeedPolicy = "changeFeedPolicy",
+  AutoPilotThroughput = "autoPilotThroughput"
+}
+
+interface UpdateStatefulValueParams {
+  key: keyof SettingsComponentState;
+  value: StatefulValuesType;
+  updateBaseline?: boolean;
+}
+
+export interface SettingsComponentProps {
+  settingsTab: SettingsTab;
+}
+
+interface SettingsComponentState {
+  throughput: StatefulValue<number>;
+  timeToLive: StatefulValue<TtlType>;
+  timeToLiveSeconds: StatefulValue<number>;
+  geospatialConfigType: StatefulValue<GeospatialConfigType>;
+  indexingPolicyContent: StatefulValue<DataModels.IndexingPolicy>;
+  conflictResolutionPolicyMode: StatefulValue<DataModels.ConflictResolutionMode>;
+  conflictResolutionPolicyPath: StatefulValue<string>;
+  conflictResolutionPolicyProcedure: StatefulValue<string>;
+  analyticalStorageTtlSelection: StatefulValue<TtlType>;
+  analyticalStorageTtlSeconds: StatefulValue<number>;
+  changeFeedPolicy: StatefulValue<ChangeFeedPolicyState>;
+  shouldDiscardIndexingPolicy: boolean;
+  indexingPolicyElementFocussed: boolean;
+  notificationStatusInfo: JSX.Element;
+  ttlOffFocused: boolean;
+  ttlOnDefaultFocused: boolean;
+  ttlOnFocused: boolean;
+  userCanChangeProvisioningTypes: boolean;
+  selectedAutoPilotTier: DataModels.AutopilotTier;
+  isAutoPilotSelected: boolean;
+  autoPilotThroughput: StatefulValue<number>;
+  wasAutopilotOriginallySet: boolean;
+  selectedTab: SettingsV2TabTypes;
+}
+
+export class SettingsComponent extends React.Component<SettingsComponentProps, SettingsComponentState> {
```

Comment:
>SettingsComponent [](start = 13, length = 17)

I believe this can be made more cleaner. Let us sync up offline.

Reply:
synced offline. removed stateful values.